src/updateConfig.ts

```diff
@@ -0,0 +1,255 @@
+}
+
+const dtConfigPath = "dt.json";
+
+function dtConfig(updatedRules: string[]): Config.IConfigurationFile {
+    const config = Config.findConfiguration(dtConfigPath).results;
+    if (!config) {
+        throw new Error(`Could not load config at ${dtConfigPath}.`);
+    }
+    // Disable ignored or non-updated rules.
+    for (const entry of config.rules.entries()) {
+        const [rule, ruleOpts] = entry;
+        if (ignoredRules.includes(rule) || (updatedRules.length > 0 && !updatedRules.includes(rule))) {
+            ruleOpts.ruleSeverity = "off";
+        }
+    }
+    return config;
+}
+
+function updateAll(dtPath: string, config: Config.IConfigurationFile): void {
+    const packages = fs.readdirSync(path.join(dtPath, "types"));
+    for (const pkg of packages) {
+        updatePackage(path.join(dtPath, "types", pkg), config);
+    }
+}
+
+function updatePackage(pkgPath: string, baseConfig: Config.IConfigurationFile): void {
+    installDependencies(pkgPath);
+    const packages = walkPackageDir(pkgPath);
+
+    const linterOpts: ILinterOptions = {
+        fix: false,
+    };
+
+    for (const pkg of packages) {
+        const results = pkg.lint(linterOpts, baseConfig);
+        if (results.failures.length > 0) {
+            const disabledRules = disableRules(results.failures);
+            const newConfig = mergeConfigRules(pkg.config(), disabledRules, baseConfig);
+            pkg.updateConfig(newConfig);
+        }
+    }
+}
+
+function installDependencies(pkgPath: string): void {
+    if (fs.existsSync(path.join(pkgPath, "package.json"))) {
+        cp.execSync(
+            "npm install --ignore-scripts --no-shrinkwrap --no-package-lock --no-bin-links",
+            {
+                encoding: "utf8",
+                cwd: pkgPath,
+            });
+    }
+}
+
+function mergeConfigRules(
+    config: Config.RawConfigFile,
+    newRules: Config.RawRulesConfig,
+    baseConfig: Config.IConfigurationFile): Config.RawConfigFile {
+        const activeRules: string[] = [];
+        baseConfig.rules.forEach((ruleOpts, ruleName) => {
+            if (ruleOpts.ruleSeverity !== "off") {
+                activeRules.push(ruleName);
+            }
+        });
+        const oldRules: Config.RawRulesConfig = config.rules || {};
+        let newRulesConfig: Config.RawRulesConfig = {};
+        for (const rule of Object.keys(oldRules)) {
+            if (activeRules.includes(rule)) {
+                continue;
+            }
+            newRulesConfig[rule] = oldRules[rule];
+        }
+        newRulesConfig = { ...newRulesConfig, ...newRules };
+        return { ...config, rules: newRulesConfig };
+}
+
+/**
+ * Represents a package from the linter's perspective.
+ * For example, `DefinitelyTyped/types/react` and `DefinitelyTyped/types/react/v15` are different
+ * packages.
+ */
+class LintPackage {
+    private rootDir: string;
+    private files: ts.SourceFile[];
+    private program: ts.Program;
+
+    constructor(rootDir: string) {
```

Comment:
```suggestion
    constructor(private rootDir: string) {
```

and then you can drop the property declaration and `this.rootDir = rootDir`